### ISSUE TITLE
[package] lcms/2.14: build issue when cross-compiling with android-ndk on Windows

### ISSUE THREAD

**Issue by elvisdukaj** [conan-io/conan-center-index#22909]
### Description

Trying to cross-compile this library with the android ndk results in build issue. It seems that `Meson` fails to setup correctly the C and C++ compiler on Windows:

```
..\src\meson.build:1:0: ERROR: Failed running 'C:\\Users\\edukaj\\.conan2\\p\\b\\andro0a7659df7e598\\p\\bin\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\aarch64-linux-android29-clang', binary or interpreter not executable.
Possibly wrong architecture or the executable bit is not set.
```

My suspect is that the path of the toolchain in MesonToolchain should have the `.cmd` suffix on Windows.

### Package and Environment Details

* Package Name/Version: **lcms/2.14**
* Operating System+version: **Windows 10**
* Compiler+version: **android-ndk/r21e, android-ndk/r25c**
* Conan version: **conan 2.1.0**
* Python version: **Python 3.11.3**

### Conan profile

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=17
compiler.libcxx=c++_static
compiler.version=9
os=Android
os.api_level=29
[tool_requires]
*: ninja/1.11.1, cmake/3.28.1, android-ndk/r21e
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.env.virtualenv:powershell=False

### Steps to reproduce

conan create recipes/lcms/all --version=2.14 --profile=android-r21e -s build_type=Release --build=missing

### Logs

<details><summary>Click to expand log</summary>

```
======== Exporting recipe to the cache ========
lcms/2.14: Exporting package recipe: C:\qxr\repo\conan-center-index\recipes\lcms\all\conanfile.py
lcms/2.14: exports: File 'conandata.yml' found. Exporting it...
lcms/2.14: Calling export_sources()
lcms/2.14: Copied 1 '.yml' file: conandata.yml
lcms/2.14: Copied 1 '.py' file: conanfile.py
lcms/2.14: Copied 1 '.patch' file: 2.13.1-0001-fix-msvc-import-lib.patch
lcms/2.14: Exported to cache folder: C:\Users\edukaj\.conan2\p\lcmsbd4af7165f7dc\e
lcms/2.14: Exported: lcms/2.14#6d7d8c5c5358886d3101bb9d3aa86457 (2024-02-27 07:46:40 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=17
compiler.libcxx=c++_static
compiler.version=9
os=Android
os.api_level=29
[tool_requires]
*: ninja/1.11.1, cmake/3.28.1, android-ndk/r21e
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.env.virtualenv:powershell=False

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    cli
Requirements
    lcms/2.14#6d7d8c5c5358886d3101bb9d3aa86457 - Cache
Build requirements
    android-ndk/r21e#0f225f36d7e8e4d9cc5c04e77f63d5f8 - Cache
    cmake/3.28.1#67506a613afcd5b43d67d77aa44521d1 - Cache
    meson/1.2.1#70f035a280b79973126f3f4dafd31ba2 - Cache
    ninja/1.11.1#de85db1490cc53d3353a536dadf1db50 - Cache

======== Computing necessary packages ========
Requirements
    lcms/2.14#6d7d8c5c5358886d3101bb9d3aa86457:22839e0749625ad0aed4839f37bb94683f2ff433 - Build
Build requirements
    android-ndk/r21e#0f225f36d7e8e4d9cc5c04e77f63d5f8:522dcea5982a3f8a5b624c16477e47195da2f84f#4b254108e79b94a70bfe8e1c6365d1b3 - Cache
    cmake/3.28.1#67506a613afcd5b43d67d77aa44521d1:522dcea5982a3f8a5b624c16477e47195da2f84f#29e6903b5006c2b2f81003c9ebf7551b - Cache
    meson/1.2.1#70f035a280b79973126f3f4dafd31ba2:da39a3ee5e6b4b0d3255bfef95601890afd80709#51dbda0520bc1033fd5d946750e9890f - Cache
    ninja/1.11.1#de85db1490cc53d3353a536dadf1db50:723257509aee8a72faf021920c2874abc738e029#ea7e68a0bd69f24789a38173498c805b - Cache

======== Installing packages ========
android-ndk/r21e: Already installed! (1 of 5)
cmake/3.28.1: Already installed! (2 of 5)
cmake/3.28.1: Appending PATH environment variable: C:\Users\edukaj\.conan2\p\b\cmakeae1bada08307d\p\bin
cmake/3.28.1: Appending PATH environment variable: C:\Users\edukaj\.conan2\p\b\cmakeae1bada08307d\p\bin
ninja/1.11.1: Already installed! (3 of 5)
meson/1.2.1: Already installed! (4 of 5)

-------- Installing package lcms/2.14 (5 of 5) --------
lcms/2.14: Building from source
lcms/2.14: Package lcms/2.14:22839e0749625ad0aed4839f37bb94683f2ff433
lcms/2.14: Copying sources to build folder
lcms/2.14: Building your package in C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b
lcms/2.14: Calling generate()
lcms/2.14: Generators folder: C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release\conan
lcms/2.14: Generating aggregated env files
lcms/2.14: Generated aggregated env files: ['conanbuild.bat', 'conanrun.sh']
lcms/2.14: Calling build()
lcms/2.14: Apply patch (portability): Fix msvc import lib
lcms/2.14: Meson configure cmd: meson setup --cross-file "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release\conan\conan_meson_cross.ini" "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release" "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\src" -Dprefix="C:/Users/edukaj/.conan2/p/b/lcms20ffd8a0aedff/p"
lcms/2.14: RUN: meson setup --cross-file "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release\conan\conan_meson_cross.ini" "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release" "C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\src" -Dprefix="C:/Users/edukaj/.conan2/p/b/lcms20ffd8a0aedff/p"
The Meson build system
Version: 1.2.1
Source dir: C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\src
Build dir: C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release
Build type: cross build
Project name: Little-CMS
Project version: 2.14

..\src\meson.build:1:0: ERROR: Failed running 'C:\\Users\\edukaj\\.conan2\\p\\b\\andro0a7659df7e598\\p\\bin\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\aarch64-linux-android29-clang', binary or interpreter not executable.
Possibly wrong architecture or the executable bit is not set.

A full log can be found at C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release\meson-logs\meson-log.txt

lcms/2.14: ERROR:
Package '22839e0749625ad0aed4839f37bb94683f2ff433' build failed
lcms/2.14: WARN: Build folder C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release
ERROR: lcms/2.14: Error in build() method, line 70
        meson.configure()
        ConanException: Error 1 while executing
```

</details>

<details><summary>meson-log.txt</summary>

```
Build started at 2024-02-27T08:46:44.984710
Main binary: C:\qxr\qxr_env\Scripts\python.exe
Build Options: -Dprefix=C:/Users/edukaj/.conan2/p/b/lcms20ffd8a0aedff/p '--cross-file=C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release\conan\conan_meson_cross.ini'
Python system: Windows
The Meson build system
Version: 1.2.1
Source dir: C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\src
Build dir: C:\Users\edukaj\.conan2\p\b\lcms20ffd8a0aedff\b\build-release
Build type: cross build
Project name: Little-CMS
Project version: 2.14

..\src\meson.build:1:0: ERROR: Failed running 'C:\\Users\\edukaj\\.conan2\\p\\b\\andro0a7659df7e598\\p\\bin\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\aarch64-linux-android29-clang', binary or interpreter not executable.
Possibly wrong architecture or the executable bit is not set.
```

</details>

<details><summary>conan_meson_cross.ini</summary>

```
[properties]
needs_exe_wrapper = true


[constants]
preprocessor_definitions = []

[project options]
wrap_mode = 'nofallback'
bindir = 'bin'
sbindir = 'bin'
libexecdir = 'bin'
includedir = 'include'
libdir = 'lib'


[binaries]
c = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\aarch64-linux-android29-clang'
cpp = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\aarch64-linux-android29-clang++'
ld = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\aarch64-linux-android-ld.exe'



ar = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\llvm-ar'
strip = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\aarch64-linux-android-strip.exe'
as = 'C:\Users\edukaj\.conan2\p\b\andro0a7659df7e598\p\bin\toolchains\llvm\prebuilt\windows-x86_64\bin\aarch64-linux-android-as.exe'




[built-in options]
buildtype = 'release'

default_library = 'static'

b_ndebug = 'true'


backend = 'ninja'
pkg_config_path = 'C:\Users\edukaj\.conan2\p\b\lcmsfae70f2d83594\b\build-release\conan'
# C/C++ arguments
c_args = [] + preprocessor_definitions
c_link_args = []
cpp_args = [] + preprocessor_definitions
cpp_link_args = []



[build_machine]
system = 'windows'
cpu_family = 'x86_64'
cpu = 'x86_64'
endian = 'little'

[host_machine]
system = 'android'
cpu_family = 'aarch64'
cpu = 'armv8'
endian = 'little'
```

</details>

**Comment by elvisdukaj** [comment]
I found the issue in the Meson toolchain file and I am creating a PR for the fix. More information can be followed in [this PR](https://github.com/conan-io/conan/pull/15760)

**Comment by elvisdukaj** [comment]
The conan MR was approved and merged with conan `2.2.0`. This thread can now be resolved
